internal/controllers/core/kubernetesapply/reconciler_test.go

```diff
@@ -295,6 +297,55 @@ func TestBasicApplyCmd_MalformedYAML(t *testing.T) {
 	}
 }
 
+func TestBasicApplyYAML_JobComplete(t *testing.T) {
+	f := newFixture(t)
+
+	jobYAML := testyaml.JobYAML
+	entities, err := k8s.ParseYAMLFromString(jobYAML)
+	require.NoError(t, err, "Invalid JobYAML")
+	require.Len(t, entities, 1, "Expected exactly 1 Job entity")
+	require.IsType(t, entities[0].Obj, &batchv1.Job{}, "Expected exactly 1 Job entity")
+	job := entities[0].Obj.(*batchv1.Job)
+	job.SetUID(uuid.NewUUID())
+	job.Status = batchv1.JobStatus{
+		Conditions: []batchv1.JobCondition{
+			{
+				Type:   batchv1.JobComplete,
+				Status: v1.ConditionTrue,
+			},
+		},
+	}
+	f.kClient.UpsertResult = entities
+
+	ka := v1alpha1.KubernetesApply{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "a",
+		},
+		Spec: v1alpha1.KubernetesApplySpec{
+			YAML: testyaml.JobYAML,
+		},
+	}
+	f.Create(&ka)
+
+	f.MustReconcile(types.NamespacedName{Name: "a"})
+	f.MustGet(types.NamespacedName{Name: "a"}, &ka)
+
+	expected := []metav1.Condition{
+		{
+			Type:   "JobComplete",
+			Status: metav1.ConditionTrue,
+		},
+	}
+
+	assert.Equal(f.T(), ka.Status.Conditions, expected,
+		"KubernetesApply status should reflect Job completion")
+
+	// Make sure that re-reconciling doesn't clear the conditions
+	f.MustReconcile(types.NamespacedName{Name: "a"})
+	assert.Equal(f.T(), ka.Status.Conditions, expected,
```

Comment:
Should you re-get `ka` here?
```suggestion
	f.MustGet(types.NamespacedName{Name: "a"}, &ka)
	assert.Equal(f.T(), ka.Status.Conditions, expected,
```